src/components/AllProductsListSection/AllProductsListSection.tsx

```diff
@@ -50,8 +78,95 @@ export default function AllProductsListSection() {
   };
 
+  const quantityForProduct = useCallback(
+    (id: string): number => {
+      const item = cart.find((cartItem) => {
+        return cartItem.product?.id === id;
+      });
+
+      if (!item) {
+        return 0;
+      } else {
+        return item.quantity;
+      }
+    },
+    [cart]
+  );
+
+  const addProductToCart = useCallback(
+    async (product: Product) => {
+      await doPost({
+        endpoint: '/api/shopping-cart/products/',
+        body: {
+          product: product.id,
+          quantity: quantityForProduct(product.id) + 1,
+        },
+      });
+      const newCart = await doGet({ endpoint: '/api/shopping-cart/' });
+      setProductAddedToCart(product);
+      setCart(newCart['order_products']);
+      setIsOpen(true);
+    },
+    [doGet, doPost, quantityForProduct]
+  );
+
   return (
     <>
+      {productAddedToCart ? (
+        <Modal isOpen={isOpen} style={modalStyles}>
+          <div className="flex h-121 flex-col items-center justify-between px-8 py-7 md:h-63 md:w-210 md:flex-row xl:mr-8 xl:w-265">
+            <div className="relative mr-8 h-44 w-65 rounded-lg border border-green-600 xl:mr-0">
+              <Image
+                src={productAddedToCart.product_picture}
+                alt=""
+                fill={true}
+                className="h-full w-full object-contain"
+              />
+              <Image
+                src={Checkmark}
+                alt=""
+                className="absolute bottom-0 right-0 h-12 w-12"
+              />
+            </div>
+            <div className="flex h-44 w-145 flex-col items-center justify-between text-dark-violet xl:mt-4">
+              <div className="text-2xl font-bold">
+                {`${name} has been successfully added to your shopping cart!`}
+              </div>
+              <div className="text-xl">
+                You can go to the checkout or keep looking for more awesome
+                items!
+              </div>
+              <div className="flex">
+                <Button
+                  text="Go to checkout"
+                  extraClasses="h-11 w-45 mr-4"
+                  handleClick={() => {
+                    setIsOpen(false);
+                  }}
+                />
+                <Button
+                  text="Continue shopping"
+                  type="outline"
+                  extraClasses="h-11 w-45"
+                  handleClick={() => {
+                    setIsOpen(false);
+                  }}
+                />
+              </div>
+            </div>
+            <div className="relative hidden h-44 w-29 md:block">
+              <Button
+                text=""
+                image={closeModal}
+                extraClasses="absolute right-0 top-0 bg-transparent border-none"
+                handleClick={() => {
+                  setIsOpen(false);
+                }}
+              />
+            </div>
+          </div>
+        </Modal>
```

Comment:
I'd suggest moving this modal to its own component, maybe named something like `AddToCartModal` or `AddCartProductModal`